File: Source/OldCode/Archive/2005 TVA Code Library/Version/2.0/Local/Source/Tva.Phasors/IConfigurationCell.vb
'*******************************************************************************************************
'  IConfigurationCell.vb - Configuration cell interface
'  Copyright © 2005 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2005
'  Primary Developer: J. Ritchie Carroll, Operations Data Architecture [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  04/16/2005 - J. Ritchie Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

''' <summary>This interface represents the protocol independent representation of a set of configuration related data settings (typically related to a PMU).</summary>
<CLSCompliant(False)> _
Public Interface IConfigurationCell

    Inherits IChannelCell, IComparable

    Shadows ReadOnly Property Parent() As IConfigurationFrame

    Property StationName() As String

    ReadOnly Property StationNameImage() As Byte()

    ReadOnly Property MaximumStationNameLength() As Int32

    Property IDLabel() As String

    ReadOnly Property IDLabelImage() As Byte()

    ReadOnly Property IDLabelLength() As Int32

    ReadOnly Property PhasorDefinitions() As PhasorDefinitionCollection

    Property PhasorDataFormat() As DataFormat

    Property PhasorCoordinateFormat() As CoordinateFormat

    Property FrequencyDefinition() As IFrequencyDefinition

    Property FrequencyDataFormat() As DataFormat

    Property NominalFrequency() As LineFrequency

    ReadOnly Property AnalogDefinitions() As AnalogDefinitionCollection

    Property AnalogDataFormat() As DataFormat

    ReadOnly Property DigitalDefinitions() As DigitalDefinitionCollection

    ReadOnly Property FrameRate() As Int16

End Interface

File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/IConfigurationCell.cs
//*******************************************************************************************************
//  IConfigurationCell.cs
//  Copyright © 2009 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-4165
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  04/16/2009 - James R Carroll
//       Generated original version of source code.
//
//*******************************************************************************************************

using System;

namespace PCS.PhasorProtocols
{
    /// <summary>
    /// Represents a protocol independent interface representation of any kind of <see cref="IConfigurationFrame"/> cell.
    /// </summary>
    public interface IConfigurationCell : IChannelCell, IEquatable<IConfigurationCell>, IComparable<IConfigurationCell>, IComparable
    {
        /// <summary>
        /// Gets a reference to the parent <see cref="IConfigurationFrame"/> for this <see cref="IConfigurationCell"/>.
        /// </summary>
        new IConfigurationFrame Parent { get; set; }

        /// <summary>
        /// Gets or sets the station name of this <see cref="IConfigurationCell"/>.
        /// </summary>
        string StationName { get; set; }

        /// <summary>
        /// Gets the binary image of the <see cref="StationName"/> of this <see cref="IConfigurationCell"/>.
        /// </summary>
        byte[] StationNameImage { get; }

        /// <summary>
        /// Gets the maximum length of the <see cref="StationName"/> of this <see cref="IConfigurationCell"/>.
        /// </summary>
        int MaximumStationNameLength { get; }

        /// <summary>
        /// Gets or sets the ID label of this <see cref="IConfigurationCell"/>.
        /// </summary>
        string IDLabel { get; set; }

        /// <summary>
        /// Gets the binary image of the <see cref="IDLabel"/> of this <see cref="IConfigurationCell"/>.
        /// </summary>
        byte[] IDLabelImage { get; }

        /// <summary>
        /// Gets the length of the <see cref="IDLabel"/> of this <see cref="IConfigurationCell"/>.
        /// </summary>
        int IDLabelLength { get; }

        /// <summary>
        /// Gets a reference to the <see cref="PhasorDefinitionCollection"/> of this <see cref="IConfigurationCell"/>.
        /// </summary>
        PhasorDefinitionCollection PhasorDefinitions { get; }

        /// <summary>
        /// Gets or sets the <see cref="DataFormat"/> for the <see cref="IPhasorDefinition"/> objects in the <see cref="PhasorDefinitions"/> of this <see cref="IConfigurationCell"/>.
        /// </summary>
        DataFormat PhasorDataFormat { get; set; }

        /// <summary>
        /// Gets or sets the <see cref="CoordinateFormat"/> for the <see cref="IPhasorDefinition"/> objects in the <see cref="PhasorDefinitions"/> of this <see cref="IConfigurationCell"/>.
        /// </summary>
        CoordinateFormat PhasorCoordinateFormat { get; set; }

        /// <summary>
        /// Gets or sets the <see cref="IFrequenceyDefinition"/> of this <see cref="IConfigurationCell"/>.
        /// </summary>
        IFrequencyDefinition FrequencyDefinition { get; set; }

        /// <summary>
        /// Gets or sets the <see cref="DataFormat"/> of the <see cref="FrequencyDefinition"/> of this <see cref="IConfigurationCell"/>.
        /// </summary>
        DataFormat FrequencyDataFormat { get; set; }

        /// <summary>
        /// Gets or sets the nominal <see cref="LineFrequency"/> of the <see cref="FrequencyDefinition"/> of this <see cref="IConfigurationCell"/>.
        /// </summary>
        LineFrequency NominalFrequency { get; set; }

        /// <summary>
        /// Gets a reference to the <see cref="AnalogDefinitionCollection"/> of this <see cref="IConfigurationCell"/>.
        /// </summary>
        AnalogDefinitionCollection AnalogDefinitions { get; }

        /// <summary>
        /// Gets or sets the <see cref="DataFormat"/> for the <see cref="IAnalogDefinition"/> objects in the <see cref="AnalogDefinitions"/> of this <see cref="IConfigurationCell"/>.
        /// </summary>
        DataFormat AnalogDataFormat { get; set; }

        /// <summary>
        /// Gets a reference to the <see cref="DigitalDefinitionCollection"/> of this <see cref="IConfigurationCell"/>.
        /// </summary>
        DigitalDefinitionCollection DigitalDefinitions { get; }

        /// <summary>
        /// Gets the specified frame rate of this <see cref="IConfigurationCell"/>.
        /// </summary>
        short FrameRate { get; }

        /// <summary>
        /// Gets or sets the revision count of this <see cref="IConfigurationCell"/>.
        /// </summary>
        ushort RevisionCount { get; set; }
    }
}